include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp

```diff
@@ -562,9 +561,9 @@ struct buffered_piece_collection
                 }
                 // Walk through them, in reverse to insert at right index
                 int index_offset = pc.robust_turns.size() - 1;
-                for (typename std::vector<robust_turn>::const_reverse_iterator
-                        rit = pc.robust_turns.rbegin();
-                    rit != pc.robust_turns.rend();
+                for (typename boost::range_reverse_iterator<const std::vector<robust_turn> >::type
+                        rit = boost::const_rbegin(pc.robust_turns);
+                    rit != boost::const_rend(pc.robust_turns);
                     ++rit, --index_offset)
                 {
                     int const index_in_vector = 1 + rit->seg_id.segment_index - piece_segment_index;
```

Comment:
What was wrong with this code? This is standard vector code - I cannot imagine that either layout changes or std code is causing issues.

Anyway, this code will go so I don't object to the change.